src/formio.js

```diff
@@ -90,7 +91,14 @@ app.config([
 
 app.run([
   '$templateCache',
-  function($templateCache) {
+  '$rootScope',
+  'ngDialog',
+  function($templateCache, $rootScope, ngDialog) {
+    // Close all open dialogs on state change.
+    $rootScope.$on('$stateChangeStart', function() {
```

Comment:
This code was copied from ngFormBuilder.js in the builder but you're right.  It seems to assume people are using ui-router in their apps which they may not be.  Not sure it's necessary at all.  The dialog is modal and clicking outside simply closes it .  Even clicking on a tab that would normally change routes.  So you can try testing without this change.  I did and it seemed OK to me.  If we do need something like this we could try...

$scope.$on('$routeChangeStart', function(next, current) { 
   ngDialog.closeAll(false);
 });

...but I'm not sure where we'd put that or on what scope.